sdk/reservations/Azure.ResourceManager.Reservations/tests/Tests/PurchaseReservationOrderTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Reservations.Models;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Reservations.Tests
+{
+    public class PurchaseReservationOrderTests : ReservationsManagementClientBase
+    {
+        private TenantResource Tenant { get; set; }
+        private ReservationOrderResponseCollection collection { get; set; }
+
+        public PurchaseReservationOrderTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public async Task ClearAndInitialize()
+        {
+            if (Mode == RecordedTestMode.Record || Mode == RecordedTestMode.Playback)
+            {
+                await InitializeClients();
+
+                AsyncPageable<TenantResource> tenantResourcesResponse = ArmClient.GetTenants().GetAllAsync();
+                List<TenantResource> tenantResources = await tenantResourcesResponse.ToEnumerableAsync();
+                Tenant = tenantResources.ToArray()[0];
+                collection = Tenant.GetReservationOrderResponses();
+            }
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestPurchaseReservationOrderSharedScopeMonthly()
+        {
+            var purchaseRequestContent = CreatePurchaseRequestContent("Shared", "Monthly");
+            var response = await Tenant.CalculateReservationOrderAsync(purchaseRequestContent);
+            var purchaseResponse = await collection.CreateOrUpdateAsync(WaitUntil.Completed, response.Value.Properties.ReservationOrderId, purchaseRequestContent);
+
+            await purchaseResponse.WaitForCompletionAsync();
```

Comment:
You don't need to wait by yourself, we will [call this method](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/reservations/Azure.ResourceManager.Reservations/src/Generated/ReservationOrderResponseCollection.cs#L80-L81) for you as you specified the `WaitUntil.Completed` in the `CreateOrUpdate` method, so please remove this line in this file.